Subject: Abort processing of crontabs with errors
Author: Faidon Liambotis <FIXME>
Last-Update: 2010-04-11

Abort processing of a crontab when an error is encountered.

Strictly speaking, this fix is mostly relevant only to the /etc/cron.d feature
patch (in order to prevent a security issue). It is include in fixes/, however,
because it made general sense. 

Bug-Debian: http://bugs.debian.org/378153

Acked-by: Christian Kastner <debian@kvr.at>
Index: patched/user.c
===================================================================
--- patched.orig/user.c	2010-05-06 18:18:27.646435759 +0200
+++ patched/user.c	2010-05-06 18:18:32.328436004 +0200
@@ -97,6 +97,9 @@
  			 * newline, so we bail out
 			 */
 			if (envstr[0] != '\0') {
+                                log_it(u->name, getpid(), "ERROR", "Missing "
+                                "newline before EOF, this crontab file will be "
+                                "ignored");
 				free_user(u);
 				u = NULL;
 			}
@@ -106,6 +109,11 @@
 			if (e) {
 				e->next = u->crontab;
 				u->crontab = e;
+			} else {
+				/* stop processing on syntax error */
+				free_user(u);
+				u = NULL;
+				goto done;
 			}
 			break;
 		case TRUE:
